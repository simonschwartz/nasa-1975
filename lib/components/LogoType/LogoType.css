.NASA1975-logo-type--lg > .NASA1975-logo-type-logo {
  max-width: 300px;
  height: auto;
}

.NASA1975-logo-type--md > .NASA1975-logo-type-logo {
  max-width: 270px;
  height: auto;
}

.NASA1975-logo-type--sm > .NASA1975-logo-type-logo {
  max-width: 190px;
  height: auto;
}

@media (min-width: 768px) {
  .NASA1975-logo-type--lg > .NASA1975-logo-type-logo {
    max-width: 380px;
    height: auto;
  }

  .NASA1975-logo-type--md > .NASA1975-logo-type-logo {
    max-width: 300px;
    height: auto;
  }

  .NASA1975-logo-type--sm > .NASA1975-logo-type-logo {
    max-width: 225px;
    height: auto;
  }
}

.NASA1975-logo-type--dark .NASA1975-logo-type-logo * {
  fill: #000;
}

.NASA1975-logo-type--red .NASA1975-logo-type-logo * {
  fill: #e30613;
}

.NASA1975-logo-type--light .NASA1975-logo-type-logo * {
  fill: #fff;
}

.NASA1975-logo-type-heading {
  color: #000;
}

.NASA1975-logo-type--light .NASA1975-logo-type-heading {
  color: #fff;
}
