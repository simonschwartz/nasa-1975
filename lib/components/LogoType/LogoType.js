import React from "react";
import PropTypes from "prop-types";
import WormLogo from "../WormLogo";
import AgencyName from "../AgencyName";
import "./LogoType.css";

const LogoType = ({
  size = "lg",
  theme = "red",
  stacked = false,
  headingLevel = "1",
  centerName = "",
  className = "",
  ...attributeOptions
}) => {
  return (
    <section
      className={`NASA1975-logo-type NASA1975-logo-type--${size} NASA1975-logo-type--${theme}`}
    >
      <WormLogo className="NASA1975-logo-type-logo" />
      <AgencyName
        size={size}
        stacked={stacked}
        className="NASA1975-logo-type-heading"
        centerName={centerName}
      />
    </section>
  );
};

LogoType.propTypes = {
  /** Stacked state of Agency name */
  stacked: PropTypes.bool,
  /** HTML heading level */
  headingLevel: PropTypes.oneOf(["1", "2", "3", "4", "5"]),
  /** Colour theme of Logo type */
  theme: PropTypes.oneOf(["dark", "light", "red"]),
  /** Size of the heading */
  size: PropTypes.oneOf(["sm", "md", "lg"]),
  /** Center name eg "John F. Kennedy Space Center" */
  centerName: PropTypes.string,
  /** className to add to heading */
  className: PropTypes.string
};

export default LogoType;
