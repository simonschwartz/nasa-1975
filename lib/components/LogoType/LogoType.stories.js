import React from "react";
import { text, select, boolean } from "@storybook/addon-knobs";
import { withInfo } from "@storybook/addon-info";
import LogoType from "./";
import { NASA_DARK } from '../../tokens';

const commonStyles = { padding: "2rem", display: 'flex', flexDirection: 'column' };
const DarkBackground = ({ children }) => <div style={{ background: NASA_DARK, ...commonStyles }}>{children}</div>;

/**
 * Docs page configuration
 */
export default {
  title: 'Logo Type',
  component: LogoType,
  parameters: {
    componentSubtitle: 'Come up with something better',
  },
};

/**
 * Component stories
 */
export const overview = () => (
  <LogoType />
);

export const darkTheme = () => (
  <LogoType theme="dark" />
);

export const lightTheme = () => (
  <DarkBackground>
    <LogoType theme="light" />
  </DarkBackground>
);

export const redTheme = () => (
  <LogoType theme="red" />
);

export const withCenterName = () => (
  <LogoType theme="dark" centerName="John F. Kennedy Space Center" />
);

export const withStackedName = () => (
  <LogoType theme="dark" stacked />
);

export const sizes = () => (
  <div>
    <LogoType theme="dark" size="lg" style={{ marginBottom: "2rem" }} />
    <LogoType theme="dark" size="md" style={{ marginBottom: "2rem" }} />
    <LogoType theme="dark" size="sm" />
  </div>
);

