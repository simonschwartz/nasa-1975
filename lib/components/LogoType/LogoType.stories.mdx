import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import LogoType from './';
import { NASA_DARK } from '../../tokens';
import StoryContainer from '../../../.storybook/components/StoryContainer';

<Meta title="Logo Type" component={LogoType} />

# Logo Type

To identify the agency, as a total entity, the NASA logotype is shown in conjunction with the full
agency name (National Aeronautics and Space Administration) as shown below.

<div style={{ margin: '3rem 0' }}>
    <LogoType theme="dark" size="lg" style={{ marginBottom: '2rem' }} />
    <LogoType theme="dark" size="md" style={{ marginBottom: '2rem' }} />
    <LogoType theme="dark" size="sm" />
</div>

## Use with color

### White Background

Against a white background the logoÂ­type may be shown in NASA red or black.

<StoryContainer>
    <LogoType theme="dark" />
</StoryContainer>

<StoryContainer>
    <LogoType theme="red" />
</StoryContainer>

### Very Light Value Background

Against a very light background, the logotype should be shown in black.

The one exception to this would be the use of NASA red logotype in very light areas of four-color
process reproduction.

<StoryContainer background="medium">
    <LogoType theme="dark" />
</StoryContainer>

<StoryContainer background="medium">
    <LogoType theme="red" />
</StoryContainer>

### Black or very dark background

Against a black or very dark color background, the logotype should always be shown in white .

The logotype should always be shown in white against a background of NASA red. The logotype should
never be shown in NASA red against a black or very dark background.

<StoryContainer background="dark">
    <LogoType theme="light" />
</StoryContainer>

<StoryContainer background="red">
    <LogoType theme="light" />
</StoryContainer>

### Medium-Value Background

Against a medium-value background, the logotype may be shown in either black or white, depending on
which is more appropriate.

The logotype should never be shown in NASA red against a medium-value background.

<StoryContainer background="grey">
    <LogoType theme="light" />
</StoryContainer>

<StoryContainer background="grey">
    <LogoType theme="dark" />
</StoryContainer>

### Center Identification

The NASA logotype and it's attendant elements of style are always a required element in any NASA
publication.

To identify any of the NASA centers the NASA logotype and full agency name is shown in conjunction
with the full center name (John F. Kennedy Space Center) as shown below.

The bold lettering style assures that the center name receives primary emphasis even though it is
always preceded by the agency name and accompanied by the NASA logo-type.

<StoryContainer>
    <LogoType theme="dark" centerName="John F. Kennedy Space Center" />
</StoryContainer>

<StoryContainer>
    <LogoType theme="red" centerName="John F. Kennedy Space Center" />
</StoryContainer>

<StoryContainer background="dark">
    <LogoType theme="light" centerName="John F. Kennedy Space Center" />
</StoryContainer>

### API

```javascript
import { LogoType } from 'nasa-1975';
```

<Props of={LogoType} />

Any other properties supplied will be spread to the root element.
