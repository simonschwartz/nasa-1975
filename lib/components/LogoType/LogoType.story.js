import React from "react";
import { text, select, boolean } from "@storybook/addon-knobs";
import { withInfo } from "@storybook/addon-info";
import LogoType from "./";

const LogoTypeStory = withInfo({
  text:
    "The NASA logotype is shown in conjunction with the full agency name. The size of the agency name should relate to the size of the logo."
})(() => {
  // Setup knob for setting stacked state
  const labelStacked = "stacked";
  const defaultValueStacked = false;
  const stackedValue = boolean(labelStacked, defaultValueStacked);

  // Setup knob for setting heading level
  const headingLevelLevel = "headingLevel";
  const optionsHeadingLevel = {
    1: "1",
    2: "2",
    3: "3",
    4: "4",
    5: "5"
  };
  const defaultValueHeadingLevel = "1";
  const headingLevelValue = select(
    headingLevelLevel,
    optionsHeadingLevel,
    defaultValueHeadingLevel
  );

  // Setup knob for setting heading size
  const labelSize = "size";
  const optionsSize = {
    lg: "lg",
    md: "md",
    sm: "sm"
  };
  const defaultValueSize = "lg";
  const sizeValue = select(labelSize, optionsSize, defaultValueSize);

  // Setup knob for setting theme
  const labelTheme = "theme";
  const optionsTheme = {
    red: "red",
    dark: "dark",
    light: "light"
  };
  const defaultValueTheme = "red";
  const themeValue = select(labelTheme, optionsTheme, defaultValueTheme);

  // Setup knob for setting center name text
  const labelCenterName = "centerName";
  const defaultValueCenterName = "";
  const centerNameValue = text(labelCenterName, defaultValueCenterName);

  return (
    <LogoType
      stacked={stackedValue}
      headingLevel={headingLevelValue}
      size={sizeValue}
      centerName={centerNameValue}
      theme={themeValue}
    />
  );
});

export default LogoTypeStory;
