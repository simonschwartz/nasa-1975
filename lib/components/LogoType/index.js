import LogoType from "./LogoType";

export default LogoType;
