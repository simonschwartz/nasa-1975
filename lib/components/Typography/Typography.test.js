import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import LogoType from "./";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<LogoType />, div);
});
