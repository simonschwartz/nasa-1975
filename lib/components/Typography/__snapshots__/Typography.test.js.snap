// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading component should correctly spread additional props to root element 1`] = `
.emotion-0 {
  color: #000000;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 2rem;
  line-height: 1;
  margin: 0;
}

<h1
  className="emotion-0"
  style={
    Object {
      "width": "200px",
    }
  }
>
  One giant leap for mankind
</h1>
`;

exports[`Heading component should render correctly given incorrect prop values 1`] = `
.emotion-0 {
  color: #000000;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 2rem;
  line-height: 1;
  margin: 0;
}

<h1
  className="emotion-0"
>
  Houston we have a problem
</h1>
`;

exports[`Heading component should render correctly given no children 1`] = `null`;

exports[`Heading component should render correctly given no props 1`] = `
.emotion-0 {
  color: #000000;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 2rem;
  line-height: 1;
  margin: 0;
}

<h1
  className="emotion-0"
>
  One giant leap for mankind
</h1>
`;

exports[`Heading component should render correctly given props {"size":"lg","level":2,"theme":"light"} 1`] = `
.emotion-0 {
  color: #FFFFFF;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 2rem;
  line-height: 1;
  margin: 0;
}

<h2
  className="emotion-0"
>
  One small step for man...
</h2>
`;

exports[`Heading component should render correctly given props {"size":"md","level":3,"weight":"medium"} 1`] = `
.emotion-0 {
  color: #000000;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 1.75rem;
  line-height: 1;
  margin: 0;
}

<h3
  className="emotion-0"
>
  One small step for man...
</h3>
`;

exports[`Heading component should render correctly given props {"size":"sm","level":4,"theme":"dark"} 1`] = `
.emotion-0 {
  color: #000000;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 1.25rem;
  line-height: 1;
  margin: 0;
}

<h4
  className="emotion-0"
>
  One small step for man...
</h4>
`;

exports[`Heading component should render correctly given props {"size":"xl","level":1,"weight":"light"} 1`] = `
.emotion-0 {
  color: #000000;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-weight: 300;
  font-size: 2.5rem;
  line-height: 1;
  margin: 0;
}

<h1
  className="emotion-0"
>
  One small step for man...
</h1>
`;

exports[`Heading component should render correctly given props {"size":5,"weight":"medium"} 1`] = `
.emotion-0 {
  color: #000000;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 2rem;
  line-height: 1;
  margin: 0;
}

<h1
  className="emotion-0"
>
  One small step for man...
</h1>
`;

exports[`Text component should correctly spread additional props to root element 1`] = `
.emotion-0 {
  color: #000000;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-weight: 300;
  font-size: 1.25rem;
  line-height: 1;
  margin: 0;
  margin-bottom: 1rem;
}

<p
  className="emotion-0"
  style={
    Object {
      "width": "200px",
    }
  }
>
  One giant leap for mankind
</p>
`;

exports[`Text component should render correctly given incorrect prop values 1`] = `
.emotion-0 {
  color: #000000;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 1.25rem;
  line-height: 1;
  margin: 0;
  margin-bottom: 1rem;
}

<houston
  className="emotion-0"
>
  Houston we have a problem
</houston>
`;

exports[`Text component should render correctly given no children 1`] = `null`;

exports[`Text component should render correctly given no props 1`] = `
.emotion-0 {
  color: #000000;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-weight: 300;
  font-size: 1.25rem;
  line-height: 1;
  margin: 0;
  margin-bottom: 1rem;
}

<p
  className="emotion-0"
>
  One giant leap for mankind
</p>
`;

exports[`Text component should render correctly given props {"size":"lg","tag":"em","theme":"light"} 1`] = `
.emotion-0 {
  color: #FFFFFF;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-weight: 300;
  font-size: 2rem;
  line-height: 1;
  margin: 0;
  margin-bottom: 1rem;
}

<em
  className="emotion-0"
>
  One small step for man...
</em>
`;

exports[`Text component should render correctly given props {"size":"md","tag":"p","weight":"medium"} 1`] = `
.emotion-0 {
  color: #000000;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 1.75rem;
  line-height: 1;
  margin: 0;
  margin-bottom: 1rem;
}

<p
  className="emotion-0"
>
  One small step for man...
</p>
`;

exports[`Text component should render correctly given props {"size":"sm","tag":"Section","theme":"dark"} 1`] = `
.emotion-0 {
  color: #000000;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-weight: 300;
  font-size: 1.25rem;
  line-height: 1;
  margin: 0;
  margin-bottom: 1rem;
}

<section
  className="emotion-0"
>
  One small step for man...
</section>
`;

exports[`Text component should render correctly given props {"size":"xl","tag":"span","weight":"light"} 1`] = `
.emotion-0 {
  color: #000000;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-weight: 300;
  font-size: 2.5rem;
  line-height: 1;
  margin: 0;
  margin-bottom: 1rem;
}

<span
  className="emotion-0"
>
  One small step for man...
</span>
`;

exports[`Text component should render correctly given props {"size":5,"weight":"medium"} 1`] = `
.emotion-0 {
  color: #000000;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-weight: 600;
  font-size: 1.25rem;
  line-height: 1;
  margin: 0;
  margin-bottom: 1rem;
}

<p
  className="emotion-0"
>
  One small step for man...
</p>
`;
